bench/src/bulk.rs

```diff
@@ -80,34 +136,128 @@ async fn client(server_addr: SocketAddr, server_cert: quinn::Certificate) -> Res
     client_config
         .add_certificate_authority(server_cert)
         .unwrap();
+    let mut client_config = client_config.build();
+    client_config.transport = Arc::new(transport_config());
+
     let quinn::NewConnection { connection, .. } = endpoint
-        .connect_with(client_config.build(), &server_addr, "localhost")
+        .connect_with(client_config, &server_addr, "localhost")
         .unwrap()
         .await
         .context("unable to connect")?;
     trace!("connected");
 
-    let mut stream = connection
-        .open_uni()
-        .await
-        .context("failed to open stream")?;
-    const DATA: &[u8] = &[0xAB; 1024 * 1024];
     let start = Instant::now();
-    for _ in 0..1024 {
-        stream
-            .write_all(DATA)
-            .await
-            .context("failed sending data")?;
+
+    let connection = Arc::new(connection);
+    let get_connection = || connection.clone();
```

Comment:
Style nit: this feels really cheesy. Let's just move a clone into the closure, even if that means doing a dance with blocks or naming something `connection2`.

Reply:
I don't get what is recommended here. The workaround with the closure is required to get access to the connection in each stream map iteration (which happens only when the whole future is awaited). A single clone won't be sufficient.